src/components/Firestore/utils.ts

```diff
@@ -15,9 +15,24 @@
  */
 
 import { firestore } from 'firebase';
-import { FieldType } from './models';
 
-export function getFieldType(value: any): FieldType {
+import {
+  FieldType,
+  FirestoreAny,
+  FirestoreArray,
+  FirestoreMap,
+  FirestorePrimitive,
+} from './models';
+
+export function getParentPath(path: string[]) {
+  return path.slice(0, path.length - 1);
+}
+
+export function getLeafKey(path: string[]) {
```

Comment:
FYI: this is lodash's `_.last(arr)` and the type can be `(input: T[]) => T`

supernit (totally optional): call it `last`?

Reply:
Cool swapped this out for lodash.last